### PR TITLE
[Test] Reduce RunTestsInSeparateProcesses usage

### PR DESCRIPTION
**Pull request by samsonasik**
@staabm RunTestsInSeparateProcesses was used to cover error tests which possible due to duplicate file fixture on run tests. This was rolled back tried, rolled back, and re-tried, see https://github.com/rectorphp/rector-src/pulls?q=is%3Apr+RunTestsInSeparateProcesses+is%3Aclosed

Let's try ~remove~ reduce  to see if now some that no longer needed.

### CODE REVIEW

File: rules-tests/Renaming/Rector/FileWithoutNamespace/PseudoNamespaceToNamespaceRector/Fixture/phpunit_test_case_split.php.inc [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 namespace Rector\Tests\Renaming\Rector\FileWithoutNamespace\PseudoNamespaceToNamespaceRector\Fixture;
 
-class Fixture6
+class PhpunitTestCaseSplit
```
Comment (staabm) [inline rules-tests/Renaming/Rector/FileWithoutNamespace/PseudoNamespaceToNamespaceRector/Fixture/phpunit_test_case_split.php.inc:5]: Do we need a new phpstan rule which makes sure classes within fixtures are uniquely named?

**Review by staabm — APPROVED** [review]

File: rules-tests/Renaming/Rector/FileWithoutNamespace/PseudoNamespaceToNamespaceRector/Fixture/phpunit_test_case_split.php.inc (hunk 1 above)
Comment (samsonasik) [reply rules-tests/Renaming/Rector/FileWithoutNamespace/PseudoNamespaceToNamespaceRector/Fixture/phpunit_test_case_split.php.inc:5]: probably needed , sometime, phpstan still read previous class "in memory", probably as "analyzed file" cache

**Comment by samsonasik** [comment]
The error seems only on CI

**Comment by samsonasik** [comment]
Let's try use `FileCacheStorage` for `RenameClassRectorTest`

**Comment by samsonasik** [comment]
it seems using `FileCacheStorage` in `RenameClassRector` still not work, reverting.

**Comment by samsonasik** [comment]
Let's rename the class fixture to avoid duplicate use by other rule https://github.com/rectorphp/rector-src/pull/3667/commits/a6f6f85497807077cd7c38bed92c9336cc092fb7

**Comment by samsonasik** [comment]
Let's rollback `RunTestsInSeparateProcesses` on RenameClassRectorTest only for now. Others are removed.

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba I am merging it ;)
